runtime/src/main/java/org/corfudb/runtime/collections/CorfuDynamicRecord.java

```diff
@@ -73,8 +78,10 @@ public boolean equals(Object obj) {
         boolean payloadMatch = payloadTypeUrl.equals(corfuDynamicRecord.payloadTypeUrl)
                 && payload.equals(corfuDynamicRecord.payload);
 
-        // Compare the metadata.
-        boolean metadataMatch = metadata.equals(corfuDynamicRecord.metadata);
+        // Compare the metadata only if both are not null.
+        boolean metadataMatch = (metadata != null && corfuDynamicRecord.metadata != null) ?
```

Comment:
minor:
metadataMatch = metadata == corfuDynamicRecord.metadata || metadata.equals(..) will make this check simpler

Reply:
done.